### ISSUE TITLE
Trying to List Instances

### ISSUE THREAD

**Issue by hiriumi** [oracle/oci-typescript-sdk#40]
I am trying to get a list of instances. I am just starting to get into OCI SDK for JavaScript (TypeScript) so I am exploring the APIs right now.

When I try to execute the following code just to try, I do get the instances but the test doesn't seem to work for me. It succeeds no matter what. I may be missing something but I am wondering what I could do to get over the issue. It may be my lack of understandings of callbacks...

```javascript
test('gets the list of instances in a compartment.', () => {
  const provider = getProvider(CONFIG_FILEPATH, PROFILE);
  const computeClient = new core.ComputeClient({
    authenticationDetailsProvider: provider,
  });
  computeClient.region = common.Region.US_ASHBURN_1;
  const listInstancesRequest = {
    compartmentId: testConfigData.tenancy_id,
  };
  computeClient.listInstances(listInstancesRequest).then(({ items }) => {
    expect(items[0].id).toEqual('adsf');
  });
});
```

**Comment by jodoglevy** [comment]
Hi @hiriumi ,

What test framework are you using, and what "failure scenario" are you expecting the test to fail on, which it does not fail on?

My hypothesis on brief look at your code is that you only have an `expect` call within the iterator which is iterating over the `listInstances` response's instances. Therefore, if the `listInstances` response has 0 instances (items) to iterate over (for example, due to an error, or due to your specified compartment not having any instances), the function called by the iterator is never called, and so the expect call is never run.

**Comment by waruwaruwaru** [comment]
Wrapping the API call in a try/catch block may help you identify what errors you are getting. Can you provide the error message that you're getting?

**Comment by ricardo-kagawa** [comment]
I don't know the specific test runner you are using, but most of the ones I know of need a signaling that a test needs to be awaited and (equivalently) when it is done running. Usually the test example is an async function, or equivalently it returns a `Promise`, or alternatively it provides a `done` callback as an argument that you must call after you are done testing (at the end of your last `then` callback).

Basically, your test is returning immediately after calling the API, and it is not waiting for the call to actually return a response. Your expectation is being tested after the test is done running (if at all), which may or may not throw an exception with your test runner. Either way, it will not get reported. At best, Node.js will complain that a Promise was rejected and not handled.

**Comment by hiriumi** [comment]
Thank you @jodoglevy , @waruwaruwaru and @ricardo-kagawa for replying to my question. I didn't think of async issue that might be the issue. I will dig around more and let you guys know.

**Comment by waruwaruwaru** [comment]
Hi @hiriumi, are you still getting any issues with the SDK?

**Comment by waruwaruwaru** [comment]
@hiriumi I will close this issue as I have not heard back. Please feel free the re-open this issue if is an SDK issue.
